File: mobile/src/api/notifications.api.ts
import apiClient from './client';

export interface NotificationItem {
  _id: string;
  title: string;
  message: string;
  type?: 'task' | 'leave' | 'attendance' | 'system' | 'general';
  isRead: boolean;
  link?: string;
  createdAt: string;
}

export const NotificationApi = {
  async getNotifications(): Promise<{ success: boolean; data: NotificationItem[]; unreadCount: number }> {
    const { data } = await apiClient.get('/api/notifications');
    return data;
  },

  async markAsRead(notificationId: string): Promise<{ success: boolean }> {
    const { data } = await apiClient.patch(`/api/notifications/${notificationId}/read`);
    return data;
  },

  async markAllAsRead(): Promise<{ success: boolean }> {
    const { data } = await apiClient.post('/api/notifications/read-all');
    return data;
  },
};

File: mobile_flutter/lib/features/notifications/data/notifications_api.dart
import '../../../core/network/api_client.dart';
import '../../../core/network/api_endpoints.dart';
import '../../../models/notification_item.dart';

class NotificationListResponse {
  final List<NotificationItem> items;
  final int unreadCount;

  NotificationListResponse({required this.items, required this.unreadCount});
}

class NotificationsApi {
  final ApiClient _client = ApiClient();

  Future<NotificationListResponse> getNotifications() async {
    final res = await _client.dio.get(ApiEndpoints.notifications);
    final data = res.data;
    final list = (data['data'] ?? data) as List;
    final items = list.map((e) => NotificationItem.fromJson(e)).toList();
    final unread = (data['unreadCount'] as num?)?.toInt() ??
        items.where((i) => !i.isRead).length;

    return NotificationListResponse(items: items, unreadCount: unread);
  }

  Future<bool> markAsRead(String notificationId) async {
    final res = await _client.dio.patch(
      ApiEndpoints.markNotificationRead(notificationId),
    );
    return res.data?['success'] ?? true;
  }

  Future<bool> markAllAsRead() async {
    final res = await _client.dio.post(ApiEndpoints.markAllNotificationsRead);
    return res.data?['success'] ?? true;
  }
}
